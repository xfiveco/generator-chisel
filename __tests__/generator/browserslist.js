const fs = require('fs-extra');

async function testWithBrowsers(browsers) {
  await global.chiselTestHelpers.generateProjectWithAnswers(
    [
      'create',
      '--skip-dependencies-install',
      '--skip-wp-download',
      '--skip-wp-config',
      '--skip-wp-install',
      '--skip-wp-plugins',
      '--skip-wp-commands',
      '--skip-format-and-build',
    ],
    [
      null,
      {
        app: {
          name: 'Chisel Test',
          author: 'Jakub Bogucki',
          projectType: 'wp-with-fe',
          browsers,
        },
      },
      {
        wp: {
          title: 'Chisel Test',
          url: 'http://chisel-test.test/',
          adminUser: 'admin',
          adminPassword: 'a',
          adminEmail: 'jakub.bogucki+chisel-test@xfive.co',
        },
      },
      { wpPlugins: { plugins: [] } },
    ],
  );

  expect(fs.readFileSync('./.browserslistrc', 'utf8')).toMatchSnapshot();
}

describe('Generator: .browserslist generation', () => {
  test('should generate empty', async () => {
    await testWithBrowsers([]);
  });

  test('should generate modern only', async () => {
    await testWithBrowsers(['modern']);
  });

  test('should generate modern & edge', async () => {
    await testWithBrowsers(['modern', 'edge18']);
  });

  test('should generate all', async () => {
    await testWithBrowsers(['modern', 'edge18', 'ie11']);
  });
});
