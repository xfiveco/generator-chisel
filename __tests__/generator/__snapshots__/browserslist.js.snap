// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Generator: .browserslist generation should generate all 1`] = `
"ie 11
edge 18
last 3 Android versions
last 3 ChromeAndroid versions
last 3 Chrome versions
last 3 Firefox versions
last 3 Safari versions
last 3 iOS versions
last 3 Edge versions
"
`;

exports[`Generator: .browserslist generation should generate empty 1`] = `
"
"
`;

exports[`Generator: .browserslist generation should generate modern & edge 1`] = `
"edge 18
last 3 Android versions
last 3 ChromeAndroid versions
last 3 Chrome versions
last 3 Firefox versions
last 3 Safari versions
last 3 iOS versions
last 3 Edge versions
"
`;

exports[`Generator: .browserslist generation should generate modern only 1`] = `
"last 3 Android versions
last 3 ChromeAndroid versions
last 3 Chrome versions
last 3 Firefox versions
last 3 Safari versions
last 3 iOS versions
last 3 Edge versions
"
`;
